src/Factories/FunFunFactory.php

```diff
@@ -922,48 +956,29 @@ private function newDonationConfirmationMailer(): DonationConfirmationMailer {
 					'mail_subject_confirm_donation',
 					'mail_subject_confirm_donation_promise'
 				)
-			)
-		);
-	}
-
-	public function newPayPalUrlGeneratorForDonations(): PayPalUrlGenerator {
-		return new PayPalUrlGenerator(
-			$this->getPayPalUrlConfigForDonations(),
-			$this->getPaymentProviderItemsTranslator()->trans( 'paypal_item_name_donation' )
-		);
-	}
-
-	public function newPayPalUrlGeneratorForMembershipApplications(): PayPalUrlGenerator {
-		return new PayPalUrlGenerator(
-			$this->getPayPalUrlConfigForMembershipApplications(),
-			$this->getPaymentProviderItemsTranslator()->trans( 'paypal_item_name_membership' )
+			),
+			$this->newGetPaymentUseCase()
 		);
 	}
 
 	private function getPayPalUrlConfigForDonations(): PayPalConfig {
 		return PayPalConfig::newFromConfig(
-			array_merge( $this->config['paypal-donation'], [ 'locale' => $this->getLocale() ] )
+			array_merge( $this->config['paypal-donation'], [ 'locale' => $this->getLocale() ] ),
+			new TranslatablePaymentItemDescription( 'paypal_item_name_donation', $this->getPaymentProviderItemsTranslator() )
 		);
 	}
 
-	private function getPayPalUrlConfigForMembershipApplications(): PayPalConfig {
-		return PayPalConfig::newFromConfig(
-			array_merge( $this->config['paypal-membership'], [ 'locale' => $this->getLocale() ] )
-		);
-	}
-
-	public function newSofortUrlGeneratorForDonations(): SofortUrlGenerator {
+	private function getSofortConfigForDonations(): SofortConfig {
 		$config = $this->config['sofort'];
 		$locale = \Locale::parseLocale( $this->getLocale() );
-		return new SofortUrlGenerator(
-			new SofortConfig(
-				$this->getPaymentProviderItemsTranslator()->trans( 'sofort_item_name_donation' ),
-				strtoupper( $locale['language'] ),
-				$config['return-url'],
-				$config['cancel-url'],
-				$config['notification-url']
-			),
-			$this->getSofortClient()
+		$translator = $this->getPaymentProviderItemsTranslator();
+		return new SofortConfig(
+			$translator->trans( 'sofort_item_name_donation' ),
+			strtoupper( $locale['language'] ),
+			$config['return-url'],
+			$config['cancel-url'],
+			$config['notification-url'],
+			new TranslatablePaymentItemDescription( 'sofort_item_name_donation', $translator )
```

Comment:
I found using the same translation key twice a bit confusing and made suggestion to improve things: https://github.com/wmde/fundraising-payments/pull/111
This would break the constructor here